### PR TITLE
[browser] JSImport binding improvements

### PR DESCRIPTION
**Pull request by pavelsavara**
Preparation for better MT dispatch.

- make `function_name` and `module_name` part of the import signature buffer
- assign import handle in C# side, not is JS side
- detect if method are async with `signature.IsAsync`
- detect if parameters have thread affinity with `signature.IsThreadCaptured` 
- made all callbacks to `task.ContinueWith` static

Other
- rename `JSSynchronizationContext.TargetThreadId` to `TargetTID` 
- rename `BindJSFunction` to `BindJSImport`
- rename `InvokeImport` to `InvokeJSImport`
- rename `mono_wasm_bind_js_function` to `mono_wasm_bind_js_import`
- rename `mono_wasm_invoke_import` to `mono_wasm_invoke_js_import`
- drop parameter in `InstallWebWorkerInterop`, always install full 
- `<EmitCompilerGeneratedFiles>` in debug mode for few projects

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs [hunk 1]
```diff
@@ -302,30 +357,27 @@ public static void LoadSatelliteAssembly(byte[] dllBytes)
...
                 }
-                else if (ctx.TargetThreadId != currentThreadId)
-                {
-                    Environment.FailFast($"JSSynchronizationContext.Install failed has wrong native thread id {ctx.TargetThreadId} != {currentThreadId}");
-                }
-                ctx.AwaitNewData();
             }
+            else if (ctx.TargetTID != currentTID)
+            {
+                Environment.FailFast($"JSSynchronizationContext.Install failed has wrong native thread id {ctx.TargetTID} != {currentTID}");
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs:378]: nit
```suggestion
                Environment.FailFast($"JSSynchronizationContext.Install has wrong native thread id {ctx.TargetTID} != {currentTID}");
```

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs [hunk 2]
```diff
@@ -21,7 +23,7 @@ internal sealed class JSSynchronizationContext : SynchronizationContext
     {
         private readonly Action _DataIsAvailable;// don't allocate Action on each call to UnsafeOnCompleted
         public readonly Thread TargetThread;
-        public readonly IntPtr TargetThreadId;
+        public readonly IntPtr TargetTID;
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs:26]: What is the motivation here? IMO the original name was easier to read

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs [hunk 3]
```diff
@@ -165,28 +164,28 @@ internal void ToJSDynamic(Task? value)
...
+            task.ContinueWith(Complete, taskHolder, TaskScheduler.FromCurrentSynchronizationContext());
 #else
-            task.GetAwaiter().OnCompleted(Complete);
+            task.ContinueWith(Complete, taskHolder, TaskScheduler.Current);
 #endif
 
-            void Complete()
+            static void Complete(Task task, object? th)
             {
+                var _taskHolder = (JSObject)th!;
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs:175]: nit: Local vars should not start with `_`

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs [hunk 4]
```diff
@@ -235,20 +234,21 @@ public void ToJS(Task? value)
...
+            task.ContinueWith(Complete, taskHolder, TaskScheduler.FromCurrentSynchronizationContext());
 #else
-            task.GetAwaiter().OnCompleted(Complete);
+            task.ContinueWith(Complete, taskHolder, TaskScheduler.Current);
 #endif
 
-            void Complete()
+            static void Complete(Task task, object? th)
             {
+                JSObject _taskHolder = (JSObject)th!;
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs:244]: nit: Local vars should not start with `_`

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs (hunk 2 above)
Comment (pavelsavara) [reply src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs:26]: Mono code uses `tid` consistently for thread id of this type. That's `MonoNativeThreadId` or `pthread_t`

As opposed to `Thread.CurrentThread.ManagedThreadId` for example. I wanted to separate them clearly.

**Review by maraf — APPROVED** [review]
Looks good to me 👍

**Pull request merged**